modules/core/src/controllers/controller.js

```diff
@@ -383,7 +376,7 @@ export default class Controller {
     const isZoomOut = this.isFunctionKeyPressed(event);
 
     const newControllerState = this.controllerState.zoom({pos, scale: isZoomOut ? 0.5 : 2});
-    return this.updateViewport(newControllerState, LINEAR_TRANSITION_PROPS);
+    return this.updateViewport(newControllerState, this._transitionProps);
```

Comment:
If you call `_getTransitionProps()` instead of caching it, the method can return different results based on the interaction state.